### PR TITLE
Fixed : Disable Submit Button 

### LINKED ISSUE #9501: Disable Submit Button on Edit Profile Page When No Changes Are Made
**Describe the bug**

On the user information edit profile page, when clicking the Submit button without making any changes to the form across multiple sections, a form update request is still triggered.

**To Reproduce**

1. Go to users page
2. Click on details button and hit edit button on all the section
3. without making any change click the submit button

**Expected behavior**

The Submit button on the user information page in edit mode should be disabled unless changes have been made to the form.

**Screenshots**

Image

### PR DESCRIPTION
**Pull request by Jeffrin2005**
## Proposed Changes

- Fixes #9501
- Submit button remains disabled until form fields are modified

https://github.com/user-attachments/assets/1bf15b34-ad1e-47e4-836e-b4cdd103fcd2

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

## Summary by CodeRabbit

- **New Features**
	- Introduced new dependencies for enhanced form functionality and internationalization.
	- Improved form state management and validation through updated logic.
	- Enhanced error handling during form submission.

- **Bug Fixes**
	- Resolved issues related to form state resetting and validation checks.

- **Documentation**
	- Updated import statements and component usage for clarity.

### CODE REVIEW

**Review by rithviknishad — CHANGES_REQUESTED** [review]
let's not use lodash. we just got rid of it from care last month.

also do use shadcn's form component instead. that'll tell if you the form is dirty or not.

**Review by bodhish — CHANGES_REQUESTED** [review]
Just noticed the review from Rithvik.

**Comment by Jeffrin2005** [comment]
@rithviknishad Implemented without using lodash

File: src/components/Form/Form.tsx [hunk 1]
```diff
@@ -55,7 +58,13 @@ const Form = <T extends FormDetails>({
       dispatch({ type: "set_form", form });
       setIsLoading(false);
     });
-  }, [asyncGetDefaults]);
+  }, [asyncGetDefaults, dispatch]);
+
+  useEffect(() => {
+    const hasChanges =
+      JSON.stringify(state.form) !== JSON.stringify(formVals.current);
```
Comment (Jacobjeevan) [inline src/components/Form/Form.tsx]: Let's not do this, comparing like this isn't always reliable.

File: src/components/Form/Form.tsx [hunk 2]
```diff
@@ -141,17 +151,22 @@ const Form = <T extends FormDetails>({
           </div>
           <div className="flex flex-col-reverse justify-end gap-3 sm:flex-row">
             {!hideCancelButton && (
-              <Cancel
+              <ButtonV2
```
Comment (Jacobjeevan) [inline src/components/Form/Form.tsx]: Use shadcn buttons instead (src/components/ui).

File: package-lock.json
Comment (Jacobjeevan) [inline package-lock.json:1]: Shouldn't be in the commit.

How are you committing your merge changes? Cli or GUI 🤔

**Review by Jacobjeevan — CHANGES_REQUESTED** [review]

File: src/components/Form/Form.tsx [hunk 3]
```diff
@@ -141,17 +151,22 @@ const Form = <T extends FormDetails>({
...
-              />
+                disabled={disabled}
+              >
+                {props.cancelLabel ?? t("common:cancel")}
+              </ButtonV2>
             )}
-            <Submit
-              data-testid="submit-button"
+            <ButtonV2
+              variant="primary"
```
Comment (Jacobjeevan) [inline src/components/Form/Form.tsx]: Make sure the data-testid is not being used in tests first, before removing them.

**Review by Jacobjeevan — CHANGES_REQUESTED** [review]

File: package-lock.json
Comment (Jeffrin2005) [reply package-lock.json:1]: CLI

File: src/components/Form/Form.tsx (hunk 1 above)
Comment (rithviknishad) [reply src/components/Form/Form.tsx]: like already said in previous review, use shadcn's form

https://ui.shadcn.com/docs/components/form

**Comment by Jeffrin2005** [comment]
@rithviknishad Used Shadcn Form and added form.tsx

File: src/components/Form/Form.tsx [hunk 4]
```diff
@@ -43,45 +47,61 @@ const Form = <T extends FormDetails>({
   hideCancelButton = false,
   ...props
 }: Props<T>) => {
+  const { t } = useTranslation();
   const initial = { form: props.defaults, errors: {} };
   const [isLoading, setIsLoading] = useState(!!asyncGetDefaults);
   const [state, dispatch] = useAutoSaveReducer<T>(formReducer, initial);
+  const [isEdited, setIsEdited] = useState(false);
```
Comment (Jacobjeevan) [inline src/components/Form/Form.tsx]: https://www.react-hook-form.com/api/useform/formstate/

use the in-built `isDirty` attribute instead.

**Review by Jacobjeevan — CHANGES_REQUESTED** [review]

File: src/components/Form/Form.tsx
Comment (rithviknishad) [inline src/components/Form/Form.tsx:1]: lets not modify the old custom `Form` component with new `Form` component from shadcn ui.

Shadcn's form is built to be used with shadcn's form fields, not our form field components.

**Review by rithviknishad — CHANGES_REQUESTED** [review]

**Comment by rithviknishad** [comment]
no longer relevant in new architecture as we are using shadcn's form with react hook form

**Pull request closed without merging**